tests/test_TopicNaming.py

```diff
@@ -0,0 +1,136 @@
+	and the topic name.  The name begins with a lowercase letter and contains
+	only alphanumeric and underscore characters.
+
+	This tests the first root, the CSC name.
+
+	Parameters
+	----------
+	xmlfile : `pathlib.Path`
+		Full filepath to the Commands or Events XML file for the CSC.
+	csc : `test_utils.subsystems`
+		Name of the CSC
+	topic : `xmlfile.stem`
+		One of ['Commands','Events','Telemetry']
+	"""
+	saltype = "SAL" + topic.rstrip('s')
+	# Check for known issues.
+	jira = check_for_issues(csc, topic, "csc")
+	if jira:
+		pytest.skip(jira + ": " + str(xmlfile.name) + " <EFDB_Topic> is not properly formed.")
+	# Test the topic <EFDB_Name> field.
+	with open(str(xmlfile), "r", encoding="utf-8") as f:
+		tree = ET.parse(f)
+	root = tree.getroot()
+	for name in root.findall(f"./{saltype}/EFDB_Topic"):
+		index = 0
+		assert name.text.split('_')[index] == csc, "<EFDB_Topic> " + name.text + \
+			' in ' + str(xmlfile.name) + ' does not properly contain the CSC name.'
+
+
+@pytest.mark.parametrize("xmlfile,csc,topic", ts_xml.get_xmlfile_csc_topic())
+def test_topic_naming_type(xmlfile, csc, topic):
+	"""Test that the <EFDB_Topic> field for topics is properly formed.
+	The <EFDB_Topic> is a compound word comprising the CSC Name, the topic type
+	and the topic name.  The name begins with a lowercase letter and contains
+	only alphanumeric and underscore characters.
+
+	This tests the second root, the topic type.
+	NOTE: Telemetry topics do not explicitly define the topic type.
+
+	Parameters
+	----------
+	xmlfile : `pathlib.Path`
+		Full filepath to the Commands or Events XML file for the CSC.
+	csc : `test_utils.subsystems`
+		Name of the CSC
+	topic : `xmlfile.stem`
+		One of ['Commands','Events','Telemetry']
+	"""
+	saltype = "SAL" + topic.rstrip('s')
+	# Check for known issues.
+	jira = check_for_issues(csc, topic, "type")
+	if jira:
+		pytest.skip(jira + ": " + str(xmlfile.name) + " <EFDB_Topic> is not properly formed.")
+	# Test the topic <EFDB_Name> field.
+	with open(str(xmlfile), "r", encoding="utf-8") as f:
+		tree = ET.parse(f)
+	root = tree.getroot()
+	for name in root.findall(f"./{saltype}/EFDB_Topic"):
+		if topic == "Telemetry":
+			# Telemetry topics do not explicitly define the topic type.
+			continue
+		elif topic == "Events":
+			topictype = "logevent"
+		else:
+			topictype = "command"
+		index = 1
+		assert name.text.split('_')[index] == topictype, \
+			"<EFDB_Topic> " + name.text + ' in ' + str(xmlfile.name) + \
+			' does not properly contain the topicType string.'
+
+
+@pytest.mark.parametrize("xmlfile,csc,topic", ts_xml.get_xmlfile_csc_topic())
+def test_topic_naming_alias(xmlfile, csc, topic):
+	"""Test that the <EFDB_Topic> field for topics is properly formed.
+	The <EFDB_Topic> is a compound word comprising the CSC Name, the topic type
+	and the topic name.  The name begins with a lowercase letter and contains
+	only alphanumeric and underscore characters.
+
+	This tests the third root, the topic name.
```

Comment:
Not quite right for telemetry. One possibility is "the last" instead of "the third".

Reply:
Consider also testing that the first element matches the CSC name and the second element (if not Telemetry) matches the topic. (Or perhaps you already do that in another test and I missed it).

Reply:
Changed "third" to "last."  The other tests are in fact done in the previous test cases of this module.